app/frontend/src/components/designer/FormViewer.vue

```diff
@@ -831,15 +833,25 @@ async function saveDraft() {
   }
 }
 
-async function sendSubmission(isDraft, sub) {
-  sub.data.lateEntry =
+async function sendSubmission(isDraft) {
+  let err = await uploadQueuedFiles();
+  if (err) {
+    return;
+  }
+
+  err = await deleteQueuedFiles();
+  if (err) {
+    return;
+  }
```

Comment:
If there are 100 files to upload, and 100 files to delete, and one of the uploads fails, then the delete don't happen, yes? Should we be trying to delete, even if the upload fails? What I mean is something like:
```suggestion
  let uploadError = await uploadQueuedFiles();
  let deleteError = await deleteQueuedFiles();
  if (uploadError || deleteError) {
    return;
  }
```

Reply:
Hmmmm, I suppose file deletions are much more reliable than file uploads, since it'll need to be an error on our end to prevent it from happening. I think your way does look much cleaner though. Either works, my biggest concern would still just be when file uploads fail. I'll update it with your changes!

Reply:
Looks great!